src/dom/DomEvent.js

```diff
@@ -48,23 +48,46 @@ var eventsKey = '_leaflet_events';
 // @alternative
 // @function off(el: HTMLElement, eventMap: Object, context?: Object): this
 // Removes a set of type/listener pairs, e.g. `{click: onClick, mousemove: onMouseMove}`
+
+// @alternative
+// @function off(el: HTMLElement, types: String): this
+// Removes all previously added listeners of given types.
+
+// @alternative
+// @function off(el: HTMLElement): this
+// Removes all previously added listeners from given HTMLElement
+function batchRemove(obj, filterFn) {
+	for (var id in obj[eventsKey]) {
+		var type = id.split(/\d/)[0];
+		if (!filterFn || filterFn(type)) {
+			removeOne(obj, type, null, null, id);
+		}
+	}
+}
+
 export function off(obj, types, fn, context) {
 
-	if (typeof types === 'object') {
+	if (arguments.length === 1) {

```

Comment:
Same as here; https://github.com/Leaflet/Leaflet/pull/7518#pullrequestreview-794001157.
I think we should use `if(!types)` instead